### ISSUE TITLE
Expiration not considered unless refresh token is present

### ISSUE THREAD

**Issue by iamcam** [AFNetworking/AFOAuth2Manager#82]
I could be mistaken, but shouldn't the expiration be set regardless the refresh token - as long as `expires_in` is present? 

It seems a recent change 73409043b6daef1cdee2b84c06456bc827cff0b1 / cfdf599faab231af798bc07a6c08d10b954e13ec made use of the expiration conditional on a refresh token being present, which isn't always the case. See also #71. The `expiration` and `refresh_token` should be considered separately, not together in an all-or-nothing fashion,

When requesting a `client_credentials` grant, I receive an access token with a limited lifetime (e.g., 1 second for testing, below), but the `AFOAuthCredential` will always tell you the token has not expired.

```
{
	"access_token": "0c06a430f3f649837f412a011cf19a0e93bd41a075cc45876db601070d862ec4",
	"token_type": "bearer",
	"expires_in": 1,
	"scope": "public"
}
```

**Comment by adc-ajindal** [comment]
Facing same issue. And due to this when application token is returned with expires_in value but no refresh_token value, expires_in value is not saved and thus we never know when to update the application token.

**Comment by mattt** [comment]
Fixed by #83. Thanks, @iamcam.
